api/v1/imageset_types.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2021 Tigera, Inc. All rights reserved.
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ImageSetSpec defines the desired state of ImageSet.
+type ImageSetSpec struct {
+	// Images is the list of images to use digests. All images that the operator will deploy
+	// must be specified.
+	Images []Image `json:"images,omitempty"`
+}
+
+type Image struct {
+	// Image is an image that the operator deploys and instead of using the built in tag
+	// the operator will use the Digest for the image identifier.
+	// The value should be the image name without registry or tag or digest.
+	// For the image `docker.io/calico/node:v3.17.1` it should be represented as `calico/node`
+	Image string `json:"image"`
+
+	// Digest is the image identifier that will be used for the Image.
+	// The field should not include a leading `@` and must be prefixed with `sha256:`, if other
+	// digest prefixes are needed please submit a feature request.
+	Digest string `json:"digest"`
+}
+
+// ImageSetStatus defines the observed state of ImageSet
+type ImageSetStatus struct {
```

Comment:
No status?

Reply:
What status would we have here? There wasn't anything in the design.
We do have a status on the Installation about which ImageSet is being used.

Reply:
I think it's fine to have no status if there's nothing meaningful to put there, but we can just delete this struct then.